src/vm/comutilnative.cpp

```diff
@@ -1238,6 +1238,35 @@ FCIMPL0(INT64, GCInterface::GetAllocatedBytesForCurrentThread)
 }
 FCIMPLEND
 
+/*===============================AllocateNewArray===============================
+**Action: Allocates a new array object. Allows passing extra flags
+**Returns: The allocated array.
+**Arguments: elementTypeHandle -> type of the element, 
+**           length -> number of elements, 
+**           clearMemory -> whether user prefer to skip clearing the content of the array.
+**Exceptions: IDS_EE_ARRAY_DIMENSIONS_EXCEEDED when size is too large. OOM if can't allocate.
+==============================================================================*/
+FCIMPL3(Object*, GCInterface::AllocateNewArray, void* arrayTypeHandle, INT32 length, CLR_BOOL clearMemory)
+{
+    CONTRACTL {
+        FCALL_CHECK;
+        PRECONDITION(length >= 0);
+    } CONTRACTL_END;
+
+    OBJECTREF pRet = NULL;
+    TypeHandle arrayType = TypeHandle::FromPtr(arrayTypeHandle);
+
+    HELPER_METHOD_FRAME_BEGIN_RET_1(pRet);
+
+    //REVIEW: ArrayNative has a number of specialized helpers like for primitive arrays. Do we want smth similar?
+    pRet = AllocateSzArray(arrayType, length, /*zeroingOptional*/ !clearMemory);
```

Comment:
Could it be a problem that in the high-level APIs this parameter is named `clearMemory`, but it's then turned into `zeroingOptional` lower down? It seems like a bit of a mis-match in expectations?

To me, `clearMemory` is more definite, but from looking at the low-level code it's clearly only a preference and the [GC doesn't respect it](https://github.com/dotnet/coreclr/pull/24096/files#diff-34369e5696b07b8796dd6939dbc69630R230) if `bContainsPointers` is true